File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/MessageConverter.java
package org.matilda.commands.types;

import com.google.protobuf.Any;
import com.google.protobuf.InvalidProtocolBufferException;
import com.google.protobuf.Message;

public class MessageConverter<T extends Message> implements ProtobufConverter<T> {
    private final Class<T> mClass;

    public MessageConverter(Class<T> clazz) {
        mClass = clazz;
    }

    @Override
    public T convertToProtobuf(T object) {
        return object;
    }

    @Override
    public T convertFromProtobuf(Any object) throws InvalidProtocolBufferException {
        return object.unpack(mClass);
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/MessageConverter.kt
package org.matilda.commands.types

import com.google.protobuf.Any
import com.google.protobuf.Message

class MessageConverter<T : Message>(private val mClass: Class<T>) : ProtobufConverter<T> {
    override fun convertToProtobuf(obj: T): T {
        return obj
    }

    override fun convertFromProtobuf(obj: Any): T {
        return obj.unpack(mClass)
    }
}
